aws/aws.go

```diff
@@ -253,3 +254,49 @@ func V2ConfigFromURLParams(ctx context.Context, q url.Values) (awsv2.Config, err
 
 	return awsv2cfg.LoadDefaultConfig(ctx, opts...)
 }
+
+// V2s3OptionsFromURLParams returns a slice of s3 options to initialize
+//  an S3 client for AWS SDK v2 initialized based on the URL  parameters in q.
+
+// It is intended to be used by URLOpeners for AWS services if
+// UseV2 returns true.
+//
+// https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3#Options
+//
+// It returns an error if q contains any unknown query parameters; callers
+// should remove any query parameters they know about from q before calling
+// V2s3OptionsFromURLParams.
+//
+// The following query options are supported:
+//   - usePathStyle: A value of "true" forces the request to use path-style addressing; sets s3v2.Options.UsePathStyle.
+//   - disableHTTPS: A value of "true" disables SSL when sending requests; sets s3v2.Options.EndpointOptions.DisableHTTPS.
```

Comment:
Please follow the existing style for URL parameters, e.g., "use_path_style" and "disable_https".

Let's add support for "no_accelerate" too? (leave the default as true, but allow people to turn it off).

Reply:
Aren't we following camelcase everywhere else?

Reference:
https://github.com/google/go-cloud/pull/3491/files#diff-6fe59643f73b478394f267669c70f00520f5902ea6d880ea1806169a2e687296L102

https://github.com/google/go-cloud/pull/3491/files#diff-6fe59643f73b478394f267669c70f00520f5902ea6d880ea1806169a2e687296L96

Reply:
Hmm, you're right. But most existing URL parameters are not camelcase, so let's follow that.

I can send a separate PR to fix those.

Reply:
Ignore my comment about "no_accelerate" -- that's already supported.